thunder/dynamo/report.py

```diff
@@ -399,6 +401,200 @@ def write_repro(
         with open(folder / file_name, "w") as f:
             print(code_str, file=f)
 
+    def _get_import_code(self, compile_fn: CompileSpecificationInterface = None, time_fn: TimerInterface = None):
+        import_strs = [
+            "\n".join(v.import_str for v in torch.fx.graph._custom_builtins.values()),
+            "\n".join(compile_fn.import_str() or []) if compile_fn else "",
+            "\n".join(time_fn.import_str() or []) if time_fn else "",
+        ]
+        return "\n".join(filter(None, import_strs))
+
+    def _get_fx_graph_class_str(self, class_name: str = "DynamoModule"):
+        return _readable(self.graph, class_name, print_output=False)
+
+    def _get_repro_code(
+        self,
+        folder,
+        compile_fn: CompileSpecificationInterface,
+        time_fn: TimerInterface,
+        serialize_inputs: bool = False,
+        inputs: Sequence[torch.Tensor | ExampleInputMetaData] = None,
+        **kwargs,
+    ):
+        from thunder.dynamo.repro_script_template import repro_bench_code_template
+
+        folder = Path(folder)
+        torch_env, thunder_pkgs = get_env()
+        class_str = textwrap.indent(self._get_fx_graph_class_str(), "    ")
+        input_str = textwrap.indent(self._get_input_str(folder, inputs, serialize_inputs), "    ")
+        extra_comment_str = kwargs.get("extra_comment_str") if "extra_comment_str" in kwargs else ""
+        import_str = self._get_import_code(compile_fn, time_fn)
+        code_str = repro_bench_code_template.format(
+            torch_env=torch_env,
+            thunder_pkgs=thunder_pkgs,
+            import_str=import_str,
+            dynamo_module=class_str,
+            inputs=input_str,
+            graph_name=self.graph_name,
+            extra_comment_str=extra_comment_str,
+        )
+        return code_str
+
+    def run_repro(
+        self,
+        compile_fn: CompileSpecificationInterface,
+        check_consistency=False,
+        inputs: Sequence[torch.Tensor | ExampleInputMetaData] = None,
+    ):
+        compiled_model = compile_fn.compile(self.graph)
+        inputs = self.get_input_metadata()
+        example_inputs = eval(f"[\n{chr(10).join(arg_like(a) for a in inputs)}]")
```

Comment:
TODO -- there's probably a nicer way to generate the example inputs that doesn't involve evaluating a string. If string evaluation is ideal, however, a comment explaining why would be nice